lib/rules/prefer-predictable-action-arguments.js

```diff
@@ -0,0 +1,57 @@
+'use strict'
+
+const getDocsUrl = require('../utils/getDocsUrl')
+
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description:
+        'suggest using "preferPredictableActionArguments": true for at the top level of machine',
+      category: 'Best Practices',
+      url: getDocsUrl('prefer-predictable-action-arguments'),
+      recommended: 'warn',
+    },
+    schema: [],
+    messages: {
+      preferPredictableActionArguments:
+        'It is advised to configure predictableActionArguments: true at the top-level of your machine config',
+    },
+  },
+
+  create: function (context) {
+    return {
+      'CallExpression[callee.name=/^createMachine$|^Machine$/] > ObjectExpression:first-child':
+        function (node) {
+          const predictableActionArgumentsProperty = node.properties.find(
+            (prop) => {
+              return prop.key.name === 'predictableActionArguments'
+            }
+          )
+
+          const hasPredictableActionArgumentsProperty = Boolean(
+            predictableActionArgumentsProperty
+          )
+
+          if (!hasPredictableActionArgumentsProperty) {
+            context.report({
+              node,
+              messageId: 'preferPredictableActionArguments',
+            })
+            return
+          }
+
+          const isPredictableActionArgumentsTruthy =
```

Comment:
It seems that optional chaining is not supported by our current .eslintrc configuration. I am getting error while trying to use optional chaining operator:
`Parsing error: Unexpected token .eslint`

I will update .eslintrc config to support optional chaining

Reply:
Hmm, I forgot that this syntax may not work for some users. I am not sure if raising the ecma version is ok, as this might prevent some users from using this eslint plugin. It is probably better doing it the old way:
```
if (!predictableActionArgumentsProperty.value || predictableActionArgumentsProperty.value.value !== true)
```
(the check that predictableActionArgumentsProperty is not null has been done earlier in the code)